processor/k8sattributesprocessor/processor.go

```diff
@@ -226,6 +226,15 @@ func (kp *kubernetesprocessor) addContainerAttributes(attrs pcommon.Map, pod *ku
 		if !ok {
 			return
 		}
+	// if there is only one container in the pod, we can fall back to that container
+	case len(pod.Containers.ByID) == 1:
+		for _, c := range pod.Containers.ByID {
+			containerSpec = c
+		}
+	case len(pod.Containers.ByName) == 1:
+		for _, c := range pod.Containers.ByName {
+			containerSpec = c
+		}
```

Comment:
Since we know the exact length of the slices can we do an implicit `[0]` instead of range? The is clearer to me that we truly are getting 1 exact value.

Reply:
Had to double check now why I did it like that in the first place :) The reason for this was that the `Containers.ByName` and `Containers.ByID` fields are maps, so just accessing the first item at index `0` is not possible here. One other approach would be to use the `maps.Keys` function of the `golang.org/x/exp/map` package, like for example:

```
case len(pod.Containers.ByName) == 1:
		containerSpec = pod.Containers.ByName[maps.Keys(pod.Containers.ByName)[0]]
```

Please let me know which option you would prefer - I don't mind, but would maybe still go with the first option, as it requires no additional import